core/unit_test/tstDeepCopy.hpp

```diff
@@ -118,6 +118,89 @@ void testDeepCopy()
     checkDataMembers( dst_aosoa, fval, dval, ival, dim_1, dim_2, dim_3 );
 }
 
+//---------------------------------------------------------------------------//
+// Perform a mirror test.
+void testMirror()
+{
+    // Data dimensions.
+    const int dim_1 = 3;
+    const int dim_2 = 2;
+    const int dim_3 = 4;
+
+    // Declare data types.
+    using DataTypes =
+        Cabana::MemberTypes<float[dim_1][dim_2][dim_3],
+                            int,
+                            double[dim_1],
+                            double[dim_1][dim_2]
+                            >;
+
+    // Create an AoSoA in the test memory space.
+    int num_data = 423;
+    Cabana::AoSoA<DataTypes,TEST_MEMSPACE> aosoa( num_data );
+
+    // Initialize data.
+    float fval = 3.4;
+    double dval = 1.23;
+    int ival = 1;
+    auto slice_0 = aosoa.template slice<0>();
+    auto slice_1 = aosoa.template slice<1>();
+    auto slice_2 = aosoa.template slice<2>();
+    auto slice_3 = aosoa.template slice<3>();
+    for ( std::size_t idx = 0; idx < aosoa.size(); ++idx )
+    {
+        // Member 0.
+        for ( int i = 0; i < dim_1; ++i )
+            for ( int j = 0; j < dim_2; ++j )
+                for ( int k = 0; k < dim_3; ++k )
+                    slice_0( idx, i, j, k ) = fval * (i+j+k);
+
+        // Member 1.
+        slice_1( idx ) = ival;
+
+        // Member 2.
+        for ( int i = 0; i < dim_1; ++i )
+            slice_2( idx, i ) = dval * i;
+
+        // Member 3.
+        for ( int i = 0; i < dim_1; ++i )
+            for ( int j = 0; j < dim_2; ++j )
+                slice_3( idx, i, j ) = dval * (i+j);
+    }
+
+    // Create a mirror with the same memory space and copy.
+    auto same_space_copy = Cabana::Experimental::create_mirror_view_and_copy(
+        TEST_MEMSPACE(), aosoa );
+    EXPECT_EQ( same_space_copy.size(), aosoa.size() );
+    bool ssc_same_ms =
+        std::is_same<TEST_MEMSPACE,
+                     decltype(same_space_copy)::memory_space>::value;
+    EXPECT_TRUE( ssc_same_ms );
+    bool ssc_same_mt =
+        std::is_same<DataTypes,
+                     decltype(same_space_copy)::member_types>::value;
+    EXPECT_TRUE( ssc_same_mt );
+
+    // Check values.
+    checkDataMembers( same_space_copy, fval, dval, ival, dim_1, dim_2, dim_3 );
```

Comment:
You need to test that there was no memory allocation.  Checking that `aosoa` and `same_space_copy` are the same type and have the same address would do.

Reply:
done

Reply:
Sure `EXPECT_EQ( aosoa.ptr(), same_space_copy.ptr() );` does check that `AoSoA` that is being returned has the same underlying data but `create_mirror_view_and_copy` states that the `AoSoA` argument is being returned which you would test with `EXPECT_EQ( &aosoa, &same_space_copy );`.